mucommander-format-xz/src/main/java/com/mucommander/commons/file/archive/xz/XzArchiveFile.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+package com.mucommander.commons.file.archive.xz;
+
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.UnsupportedFileOperationException;
+import com.mucommander.commons.file.archive.AbstractROArchiveFile;
+import com.mucommander.commons.file.archive.ArchiveEntry;
+import com.mucommander.commons.file.archive.ArchiveEntryIterator;
+import com.mucommander.commons.file.archive.SingleArchiveEntryIterator;
+
+import org.tukaani.xz.XZInputStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * XzArchiveFile provides read-only access to archives in the xz format.
+ *
+ * <p>The actual decompression work is performed by the {@link org.tukaani.xz.XZInputStream} class.</p>
+ *
+ * @see com.mucommander.commons.file.archive.xz.XzFormatProvider
+ * @author Giorgos Retsinas
+ */
+public class XzArchiveFile extends AbstractROArchiveFile {
```

Comment:
I see [here](https://commons.apache.org/proper/commons-compress/javadocs/api-1.6/org/apache/commons/compress/compressors/xz/package-summary.html) that both input and output streams exist for XZ, why making it read-only?

Reply:
I think we will need to rework RWArchiveEntryFile in order to do this. Currently, when a file inside an archive is updated it removes the existing entry and then adds back the updated one. However, this does not make sense for single-file formats like XZ. The same is true for AbstractRWArchiveFile and its methods addEntry and deleteEntry. I can take a deeper look into this eventually and then also update gzip and bzip2 format support which currently work the same way.

In the meantime, I added support for compressing files using XZ in the archiver which did not require any significant changes.